ssl/ssl_ciph.c

```diff
@@ -1945,3 +1945,55 @@ int SSL_CIPHER_is_aead(const SSL_CIPHER *c)
 {
     return (c->algorithm_mac & SSL_AEAD) ? 1 : 0;
 }
+
+int ssl_cipher_get_overhead(const SSL_CIPHER *c,
+			    unsigned int *mac_overhead,
+			    unsigned int *int_overhead,
+			    unsigned int *blocksize,
+			    unsigned int *ext_overhead)
+{
+	unsigned int mac = 0, in = 0, blk = 0, out = 0;
+
+	if (c->algorithm_enc & SSL_AESGCM) {
+		out = EVP_GCM_TLS_EXPLICIT_IV_LEN + EVP_GCM_TLS_TAG_LEN;
+	} else if (c->algorithm_enc & (SSL_AES128CCM | SSL_AES256CCM)) {
+		out = EVP_CCM_TLS_EXPLICIT_IV_LEN + 16;
+	} else if (c->algorithm_enc & (SSL_AES128CCM8 | SSL_AES256CCM8)) {
+		out = EVP_CCM_TLS_EXPLICIT_IV_LEN + 8;
+	} else if (c->algorithm_enc & SSL_CHACHA20POLY1305) {
+		out = 16;
```

Comment:
I think it uses a heuristic based on whether the line being commented upon has changed.